### PR TITLE
Sample app using :lib-dataprotect

### PR DESCRIPTION
**Pull request by linuxwolf**
This is a simple demo application that exercises portions of :lib-dataprotect

### CODE REVIEW

File: samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import android.util.Log
+import android.widget.Button
+import mozilla.components.lib.dataprotect.Keystore
+import mozilla.components.support.base.log.logger.Logger
+import java.nio.charset.StandardCharsets
+
+class MainActivity : AppCompatActivity() {
+    private val logger: Logger = Logger("dataprotect")
+    private val keystore: Keystore = Keystore(KEYSTORE_LABEL)
+    private val itemKeys: List<String> = ArrayList<String>().apply {
```
Comment (csadilek) [inline samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt]: This block can be replaced with a single line: 
```Kotlin
private val itemKeys: List<String> = List(5) { "protected item ${it+1}" }
```

File: samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        keystore.generateKey()
+
+        // setup protected data key/value pairs
+        val prefs = PreferenceManager.getDefaultSharedPreferences(this).apply {
+            prepareProtectedData(this)
+        }
+
+        // setup recycler
+        listAdapter = ProtectedDataAdapter(prefs, keystore, itemKeys)
+        listView = findViewById(R.id.protecteddata_list) as RecyclerView
```
Comment (csadilek) [inline samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt]: Cast is unnecessary. (`as RecyclerView` can be removed)

File: samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        listView.apply {
+            setHasFixedSize(true)
+            layoutManager = LinearLayoutManager(this@MainActivity)
+            adapter = listAdapter
+        }
+
+        toggleBtn = findViewById(R.id.toggle_locked_btn) as Button
```
Comment (csadilek) [inline samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt]: Cast is unnecessary. (`as Button` can be removed)

**Comment by linuxwolf** [comment]
@csadilek Ugh, just realized I forgot license headers.  I can I push one more (squashed) update to add those?

**Comment by csadilek** [comment]
@linuxwolf sure, thanks for catching that!

**Pull request merged**